update tool/generate_* scripts and sync files generated by those scripts

eEQK opened the pull request:
there are some changes I'm not sure should be there? see my comments

cc @davidmorgan

eEQK commented on tool/generate_converter:
```diff
@@ -48,7 +48,7 @@ done
 # Classes.
 for type in $(echo "$source" \
-  | egrep '^[a-z ]*class ' \
+  | grep -E '^[a-z ]*class ' \
```
I'm using linux and this is what I get when I run fgrep/egrep

```
❯ egrep
egrep: warning: egrep is obsolescent; using grep -E
Usage: grep [OPTION]... PATTERNS [FILE]...
Try 'grep --help' for more information.


❯ fgrep
fgrep: warning: fgrep is obsolescent; using grep -F
Usage: grep [OPTION]... PATTERNS [FILE]...
Try 'grep --help' for more information.
```


I did look up if there's any difference between the two and there seems to be none so I took the liberty to change it to the newer version

eEQK commented on tool/generate_converter:
```diff
@@ -7,7 +7,7 @@
 # mkdir /tmp/inputs
 # cp ~/git/dart-sdk/sdk/pkg/_fe_analyzer_shared/lib/src/metadata/* /tmp/inputs
-# rm /tmp/inputs/proto.dart /tmp/inputs/parser.dart /tmp/inputs/scope.dart
+# rm /tmp/inputs/proto.dart /tmp/inputs/parser.dart /tmp/inputs/scope.dart /tmp/inputs/evaluate.dart
```
I'm pretty sure this file was added recently and should be excluded

https://github.com/dart-lang/sdk/blob/805935df2102a8506d51c6ff0119a5b0d69074d4/pkg/_fe_analyzer_shared/lib/src/metadata/evaluate.dart#L630

eEQK commented on tool/dart_model_generator/lib/generate_dart_model.dart:
```diff
@@ -459,6 +459,11 @@ class TypeReference {
       };
+    } else if (name == 'double') {
+      return {
+        'type': 'double',
```
is this type here correct? I wasn't sure what types are available here, I suspected JS types since maybe it is used for LSP or something, but that does not seem to be the case considering `integer` a branch above :thinking:

eEQK commented on tool/dart_model_generator/lib/macro_metadata_definitions.dart:
```diff
@@ -11,14 +11,12 @@ import 'generate_dart_model.dart';
 final definitions = [
   Definition.union(
     'Argument',
-    createInBuffer: true,
```
was this added manually? after running the generate script it is gone

eEQK commented on tool/generate_converter:
```diff
@@ -65,7 +65,18 @@ for type in $(echo "$source" \
+      else
+        echo "    $field_name: convert(o.$field_name),"
+      fi
+    elif echo "$line" | grep -q ' get '; then
```
name is defined as a getter for expressions, hence the new if branch

https://github.com/dart-lang/sdk/blob/805935df2102a8506d51c6ff0119a5b0d69074d4/pkg/_fe_analyzer_shared/lib/src/metadata/references.dart#L12

eEQK commented on pkgs/_analyzer_cfe_macros/lib/metadata_converter.dart:
```diff
@@ -275,24 +281,27 @@ T? convert<T>(Object? object) => switch (object) {
   front_end.ConstructorInvocation o =>
     dart_model.ConstructorInvocation(
-          type: convertToTypeAnnotation(o.type),
+          type: convert(o.type),
```
was this added manually?

davidmorgan replied:
Yes, it's needed whenever the type is a union type, the script doesn't get this one right.

The script was not really intended for long term used, we need to build a maintainable solution :)

eEQK commented on pkgs/dart_model/lib/src/macro_metadata.g.dart:
```diff
@@ -1513,24 +1497,24 @@ extension type Instantiation.fromJson(Map<String, Object?> node)
-  });
-  IntegerLiteral({String? text})
-    : this.fromJson(Scope.createMap(_schema, text));
+  IntegerLiteral({String? text, int? value})
```
not sure why [`IntegerLiteral`](https://github.com/dart-lang/sdk/blame/f34a2ed99fc1b34cedbd974a5801f8d922121126/pkg/_fe_analyzer_shared/lib/src/metadata/expressions.dart#L93) has nullable fields but [`DoubleLiteral`](https://github.com/dart-lang/sdk/blame/f34a2ed99fc1b34cedbd974a5801f8d922121126/pkg/_fe_analyzer_shared/lib/src/metadata/expressions.dart#L108) has non-nullable? this causes some tests to fail and I'm not sure what to make of it

davidmorgan reviewed (commented):
Thanks! Looking good so far.

davidmorgan commented:
Looks good to me--is it ready for review/merge? Thanks!

davidmorgan commented:
Hmmm it should be a trivial change now I think--just check the name? Then, it's a good e2e test of the PR :)

Or if it's more work, sure, another PR sounds good.

davidmorgan commented on pkgs/dart_model/lib/src/json_buffer/json_buffer_builder.dart:
```diff
@@ -4,6 +4,7 @@
 
 import 'dart:collection';
 import 'dart:convert';
+import 'dart:io';
```
presubmits says this is unused, please remove :)

eEQK commented:
should be ok now, I've also updated the test case

davidmorgan commented on pkgs/_test_macros/lib/built_value.dart:
```diff
@@ -3,6 +3,7 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'dart:async';
+import 'dart:io';
```
Please remove this unused import also :) thanks.

eEQK replied:
sorry :weary:

davidmorgan commented:
No worries, that's what we have presubmit checks for :) let's see if they turn green this time...

eEQK commented:
Fixed the typo and updated goldens, I did run mono_repo analyze && test and I do have a single test failing in `macros/pkgs/_macro_tool` but I hope it's just a local thing because it fails on `main` for me as well

```
❯ dart test
Building package executable...
Built test:test.
00:06 +1 -1: test/macro_runner_test.dart: CfeMacroRunner runs macros [E]
  Crash when compiling:
  'package:front_end/src/source/source_loader.dart': Failed assertion: line 1482 pos 9: '_compilationUnits.values.every((compilationUnit) =>
              !(compilationUnit is SourceCompilationUnit &&
                  compilationUnit.isAugmenting))': Augmentation library found in libraryBuilders: Instance of 'DillCompilationUnitImpl', Instance of 'DillCompilationUnitImpl', ...
```

davidmorgan commented:
Looks great, thanks :)
